### ISSUE TITLE
Calling bootstrap.sh fails

### ISSUE THREAD

**Issue by Nurgak** [rock-core/buildconf#6]
I'd like to make a new ROCK installation using the bootstrap file from the [installation tutorial](http://rock-robotics.org/stable/documentation/installation.html), but I keep getting the following error, is there a known solution for that:

```
ubuntu@ubuntu:~/dev$ ./bootstrap.sh ruby2.0
Which protocol do you want to use to access rock-core/buildconf.git on github.com? [git|ssh|http] (default: http) 
Detected 'gem' to be /usr/bin/gem2.0
Detected bundler at /home/ubuntu/.autoproj/gems/ruby/2.0.0/bin/bundler
Installing autoproj in /home/ubuntu/.autoproj/gems/ruby/2.0.0

[!] There was an error parsing `Gemfile`: no implicit conversion of nil into String. Bundler cannot continue.

 #  from /home/ubuntu/dev/.autoproj/Gemfile:4
 #  -------------------------------------------
 #  gem "utilrb", ">= 3.0.1"
 >  config_path = File.join(__dir__, 'config.yml')
 #  if File.file?(config_path)
 #  -------------------------------------------
FATAL: failed to install autoproj in /home/ubuntu/dev/.autoproj
```

This is on a brand new Ubuntu 14.04 with newly installed ruby2.0

**Comment by doudou** [comment]
> "brand new" Ubuntu 14.04

Mmmmm ... Irony ;-)

Anyways. I could fix this one (you're hitting a bug fixed on "newer" patch versions of ruby 2.0), but it's more than likely that you would hit some other bugs. autoproj v2 drops support of ruby 2.0, as is Rock itself. I'm currently not in a position to support old setups like this with v2.

Not all is lost, though. You have a few solutions (from the less intrusive to the most intrusive):
 - stick with autoproj v1 on 14.04/Ruby 2.0. You simply need to bootstrap using https://raw.githubusercontent.com/rock-core/autoproj/v1/bin/autoproj_bootstrap instead of the default autoproj_bootstrap script from rock-robotics.org
 - use rbenv to install the latest 2.0 patchlevel (2.0.0-p643 bootstrapped fine)
 - use rbenv to install a newer ruby version (2.3 is a great pick, I don't yet have experience with 2.4 and Rock).
 - upgrade to Ubuntu 16.04

**Comment by Nurgak** [comment]
Thanks @doudou, I chose the first option and it almost worked. I got the following error:

`osdep ruby-dev does not exist. It is referred to by thin.: Autoproj::OSDependencies::InvalidRecursiveStatement`

So I just commented out the `ruby-dev` dependency from `autoproj/remotes/rock.core/rock.osdeps` under `thin` and it installed what it needed when I called `amake`. I don't know what implication it has but I can work again now.
